cpp/arcticdb/processing/operation_dispatch_binary.hpp

```diff
@@ -51,72 +51,67 @@ VariantData binary_membership(const ColumnWithStrings& column_with_strings, Valu
             return EmptyResult{};
         } else if constexpr(std::is_same_v<Func, IsNotInOperator&&>) {
             return FullResult{};
-        } else {
-            internal::raise<ErrorCode::E_ASSERTION_FAILURE>("Unexpected operator passed to binary_membership");
         }
     }
-    util::BitSet output_bitset(column_with_strings.column_->row_count());
-    // If the value set is empty, then for IsInOperator, we are done
-    // For the IsNotInOperator, set all bits to 1
+    // If the value set is empty, we can short-circuit
     if (value_set.empty()) {
-        if constexpr(std::is_same_v<Func, IsNotInOperator&&>)
-            output_bitset.set();
-    } else {
-        details::visit_type(column_with_strings.column_->type().data_type(),[&] (auto col_desc_tag) {
-            using col_type_info = ScalarTypeInfo<decltype(col_desc_tag)>;
-            details::visit_type(value_set.base_type().data_type(), [&] (auto val_set_desc_tag) {
-                using val_set_type_info = ScalarTypeInfo<decltype(val_set_desc_tag)>;
-                if constexpr(is_sequence_type(col_type_info::data_type) && is_sequence_type(val_set_type_info::data_type)) {
-                    std::shared_ptr<std::unordered_set<std::string>> typed_value_set;
-                    if constexpr(is_fixed_string_type(col_type_info::data_type)) {
-                        auto width = column_with_strings.get_fixed_width_string_size();
-                        if (width.has_value()) {
-                            typed_value_set = value_set.get_fixed_width_string_set(*width);
-                        }
-                    } else {
-                        typed_value_set = value_set.get_set<std::string>();
+        if constexpr(std::is_same_v<Func, IsNotInOperator&&>) {
+            return FullResult{};
+        } else {
+            return EmptyResult{};
+        }
+    }
+
+    util::BitSet output_bitset;
+    constexpr auto sparse_missing_value_output = std::is_same_v<Func, IsNotInOperator&&>;
+    details::visit_type(column_with_strings.column_->type().data_type(),[&] (auto col_tag) {
```

Comment:
Not necessarily for now, but we should provide a helper method on ColumnWithStrings to make it more readable, as it's such a common thing to do (i.e. to get the data_type()